sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -51,6 +51,31 @@ class RegexParser(pattern: String) {
     ast
   }
 
+  def parseReplacement(numCaptureGroups: Int): RegexReplacement = {
+    val sequence = RegexReplacement(new ListBuffer(), numCaptureGroups)
+    while (!eof()) {
+      parseReplacementBase(() => eof()) match {
+        case RegexSequence(parts) =>
+          sequence.parts ++= parts
+        case other =>
+          sequence.parts += other
+      }
+    }
+    sequence
+  }
+
+  def parseReplacementBase(until: () => Boolean): RegexAST = {
```

Comment:
remove unused param:

```suggestion
  def parseReplacementBase(): RegexAST = {
```